site/src/pages/TemplatesPage/TemplatesPageView.tsx

```diff
@@ -171,7 +174,17 @@ export const TemplatesPageView: FC<TemplatesPageViewProps> = ({
     <Margins>
       <PageHeader
         actions={
-          canCreateTemplates && <CreateTemplateButton onNavigate={navigate} />
```

Comment:
Can we remove the `CreateTemplateButton` file? You can run `pnpm deadcode` to find unused code in the site codebase. It may be useful since this refactoring is quite big.

Ref: https://github.com/coder/coder/blob/8d4bccc612f59531ecc19034fdca3ea012629f0f/site/package.json#L30